extensions-builtin/SwinIR/scripts/swinir_model.py

```diff
@@ -142,7 +128,16 @@ def upscale(
         return Image.fromarray(output, "RGB")
 
 
-def inference(img, model, tile, tile_overlap, window_size, scale):
+def inference(
+    img,
+    model,
+    *,
+    tile: int,
+    tile_overlap: int,
+    window_size: int,
+    scale: int,
+    device,
+):
```

Comment:
This smells like `tiled_upscale`, but with tile overlap handled by weight scaling.